README.md

```diff
@@ -42,60 +52,200 @@ use PinkCrab\Registerables\Post_Type;
+  }
+}
 
-If you would like to run the tests for this package, please ensure you add your database details into the test/wp-config.php file before running phpunit.
+/** Injected into post type */
+class My_CPT extends Post_Type {
+  public $key      = 'my_cpt';
+  public $singular = 'CPT Post';
+  public $plural   = 'CPT Posts';
+
+  // Pass the service in as a dependency.
+  private Meta_Box_Service $meta_box_service;
+  public function __construct(Meta_Box_Service $meta_box_service){
+    $this->meta_box_service = $meta_box_service;
+  }
+
+  // Return the populated Meta_Box instances.
+  public function meta_boxes( array $meta_boxes ): array {
+    return $this->meta_box_service->get_meta_boxes();
+  }
+}
+```
 
-``` bash
-$ phpunit
+[See full Meta Box Docs](docs/Meta_Box.md)
+
+### Shared Meta Boxes
+
+Sometimes you want to render the same Meta Boxes on multiple Custom Post Types or added to existing ones. We have a simple `Shared_Meta_Box_Controller` base class which can be extended to register meta boxes isolated from a Post Type.
+
+
+```php
+class Acme_Meta_Box extends Shared_Meta_Box_Controller {
+  /**
+   * Return the Meta Box instance.
+   */
+  public function meta_box(): Meta_Box {
+    return Meta_Box::side('acme_box')
+      ->label('Acme Meta Box')
+      ->screen('acme_post_type_a')
+      ->screen('acme_post_type_b')
+      ->view_template('path/to/view/template')
+      ->view_vars(['some' => 'additional data passed to view'])
+      ->action('save_post', [$this, 'save_method'])
+      ->action('update_post', [$this, 'save_method'])
+  }
+
+  /**
+   * Sets any meta data against the meta box.
+   * @see Post Type docs for more details
+   */
+  public function meta_data( array $meta_data ): array {
+    $meta_data[] = ( new Meta_Data( 'acme_meta_1' ) )
+      ->type( 'integer' )
+        ->single......
+ 
+    return $meta_data;
+  }
+
+  /** The save_post|update_post hook callback */
+  public function save_method( int $post_id ): array {
+    // Handle validating and updating post meta.
+    update_post_meta($post_id, 'acme_meta_1', $value);
+  }
+}
 ```
+**[Defining Meta Data](docs/Post-Type.md#registering-meta_data)**
 
-```bash 
-$ composer test
+> The above Meta Box would be shown on both `acme_post_type_a` and `acme_post_type_b`  
+> You can also inject any dependencies via the constructor too.
 
-``` 
+## Meta Data
 
-### PHP Stan ###
+You can register, `post`, `term`, `user` and `comment` meta fields either as a part of Post Types/Taxonomy Registerables or on there own. This fluent object based definition makes it easy to create these inline.
 
-The module comes with a polyfill for all WP Functions, allowing for the testing of all core files. The current config omits the Dice file as this is not ours. To run the suite call.
-```bash 
-$ vendor/bin/phpstan analyse src/ -l8 
-```
+Can easily add full REST support by supplying a schema for the field and Registrar will register the field also.
```

Comment:
Instead of "Can easily add" I would just say "You can add" something like 

You can add full REST support by supplying a schema for the field and the Registrar will register the field also.

Not sure what the Registrar is here